contrib/babelfishpg_tsql/src/string.c

```diff
@@ -56,11 +56,29 @@ static Datum return_varchar_pointer(char *buf, int size);
 Datum
 hashbytes(PG_FUNCTION_ARGS)
 {
-	const char *algorithm = text_to_cstring(PG_GETARG_TEXT_P(0));
-	bytea	   *in = PG_GETARG_BYTEA_PP(1);
-	size_t		len = VARSIZE_ANY_EXHDR(in);
-	const uint8 *data = (unsigned char *) VARDATA_ANY(in);
-	bytea	   *result;
+    Oid            input_type = get_fn_expr_argtype(fcinfo->flinfo,1);
+    const char *algorithm = text_to_cstring(PG_GETARG_TEXT_P(0));
```

Comment:
where is it being freed? and why const char *?

Reply:
I have not freed the data,since we are not doing any palloc for data